sdk/storage/storage-blob/src/policies/StorageBearerTokenChallengeAuthenticationPolicy.ts

```diff
@@ -0,0 +1,423 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AccessToken, GetTokenOptions, TokenCredential, URLBuilder } from "@azure/core-http";
```

Comment:
Can you see if you can make StorageBearerTokenChallengeAuthenticationPolicy.ts extends from bearerTokenAuthenticationPolicy? This way, you don't need to copy & paste duplicated code.

Reply:
bearerTokenAuthenticationPolicy is not exposed by core-http, it only expose a function to create the policy.